drivers/mpsl/clock_control/clock_control_mpsl.c

```diff
@@ -207,9 +240,45 @@ static enum clock_control_status clock_get_status(struct device *dev,
 	}
 }
 
+static int clock_async_start(struct device *dev, clock_control_subsys_t subsys,
+			     struct clock_control_async_data *data)
+{
+	struct mpsl_clock_control_data *mpsl_control_data = dev->driver_data;
+	enum clock_control_status clock_status = clock_get_status(dev, subsys);
+	sys_slist_t *list = &(mpsl_control_data->async_on_list);
+
+	switch (clock_status) {
+	case CLOCK_CONTROL_STATUS_ON:
+		/* LFCLK is started by default and handled as STATUS_ON */
+		if ((data != NULL) && (data->cb != NULL)) {
+			data->cb(dev, data->user_data);
+		}
+
+		return 0;
+	case CLOCK_CONTROL_STATUS_STARTING:
+	case CLOCK_CONTROL_STATUS_OFF:
+		/* Noop */
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	if ((data != NULL) && (data->cb != NULL)) {
+		/* if same request already scheduled and not yet completed. */
+		if (clock_list_find(list, &data->node)) {
+			return -EBUSY;
+		}
+
+		clock_list_append(list, &data->node);
```

Comment:
list is modified in the clock started event and i don't see any protection against that.

Reply:
Is irq_lock() enough in this case or any suggestion?

Reply:
yes

Reply:
ok, then the list is protected by clock_list_append as well as clock_list_find.